### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by Frezzle**
Signed-off-by: Frederico Mazzone <frederico.mazzone@citrix.com>

### CODE REVIEW

**Review by gaborigloi — APPROVED** [review]
Well written, really useful guide :)

**Review by gaborigloi — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Read through the `Vagrantfile` to understand the tools used to build and develop the Toolstack components. Then, execute all the steps which apply to your setup; for example, you may choose to only install one/some of the supported text editors, whereas the `Vagrantfile` sets up all of the available ones.
+
+## Building different components
+
+In the `Vagrantfile`, `xapi` is built in order to verify that the environment has been set up correctly. In order to build other components, their dependencies must also be installed.
+
+Using `xenopsd` as an example, its dependencies must first be installed:
+
+```
+sudo opam depext -y xenopsd
```
Comment (gaborigloi) [inline README.md:33]: `sudo` is not required here as far as I know, when I do it without `sudo` it automatically runs `sudo` when it is needed. I think it's best to run it without `sudo`.

Comment (Frezzle) [reply README.md:33]: It seems you are right, as we don't run it with sudo here either: https://github.com/xapi-project/xen-api/blob/master/.travis-opam-coverage.sh#L25

Comment (gaborigloi) [reply README.md:33]: That's because I wrote that line (and of course I agree with myself) :D

**Pull request merged**